File: edukate-notifier/src/main/java/io/github/sanyavertolet/edukate/notifier/dtos/CheckedNotificationDto.java
package io.github.sanyavertolet.edukate.notifier.dtos;

import com.fasterxml.jackson.annotation.JsonTypeName;
import io.github.sanyavertolet.edukate.common.checks.CheckStatus;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@JsonTypeName("checked")
public final class CheckedNotificationDto extends BaseNotificationDto {
    private final String submissionId;
    private final String problemId;
    private final CheckStatus status;

    public CheckedNotificationDto(
            String uuid, Boolean isRead, Instant createdAt, String submissionId, String problemId, CheckStatus status
    ) {
        super(uuid, isRead, createdAt);
        this.submissionId = submissionId;
        this.problemId = problemId;
        this.status = status;
    }
}

File: edukate-notifier/src/main/kotlin/io/github/sanyavertolet/edukate/notifier/dtos/CheckedNotificationDto.kt
package io.github.sanyavertolet.edukate.notifier.dtos

import com.fasterxml.jackson.annotation.JsonTypeName
import io.github.sanyavertolet.edukate.common.checks.CheckStatus
import java.time.Instant

@JsonTypeName("checked")
data class CheckedNotificationDto(
    override val uuid: String,
    override val isRead: Boolean,
    override val createdAt: Instant,
    val submissionId: String,
    val problemId: String,
    val status: CheckStatus,
) : BaseNotificationDto
